catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -215,8 +222,94 @@ private org.apache.hadoop.hive.metastore.api.Partition toHivePartition(
   }
 
   @Override
-  public boolean dropPartition(String partitionName) {
-    throw new UnsupportedOperationException();
+  public boolean dropPartition(String partitionName, boolean ifExists)
+      throws NoSuchPartitionException {
+    try {
+      // Get all children partitions for one parent partition name,
+      // cascade delete the parent partition of all its children partitions
+      Table hiveTable = table.clientPool().run(c -> c.getTable(table.schemaName(), table.name()));
+      List<org.apache.hadoop.hive.metastore.api.Partition> partitions =
+          table
+              .clientPool()
+              .run(
+                  c ->
+                      c.listPartitions(
+                          table.schemaName(),
+                          table.name(),
+                          getFilterPartitionValueList(hiveTable, partitionName),
+                          (short) -1));
+      if (partitions.isEmpty()) {
+        throw new NoSuchPartitionException(
+            "Hive partition %s does not exist in Hive Metastore", partitionName);
+      }
+      // Delete partitions iteratively
+      for (org.apache.hadoop.hive.metastore.api.Partition partition : partitions) {
+        table
+            .clientPool()
+            .run(
+                c ->
+                    c.dropPartition(
+                        partition.getDbName(),
+                        partition.getTableName(),
+                        partition.getValues(),
+                        false));
+      }
+    } catch (NoSuchPartitionException e) {
+      if (ifExists) {
+        return true;
+      }
+      throw new NoSuchPartitionException(
+          "Hive partition %s does not exist in Hive Metastore", partitionName);
+
+    } catch (UnknownTableException e) {
+      throw new NoSuchTableException(
+          e, "Hive table %s does not exist in Hive Metastore", table.name());
+
+    } catch (TException | InterruptedException e) {
+      throw new RuntimeException(
+          "Failed to get partition "
+              + partitionName
+              + " of table "
+              + table.name()
+              + "from Hive Metastore",
+          e);
+    }
+    return true;
+  }
+
+  private List<String> getFilterPartitionValueList(Table dropTable, String partitionSpec)
```

Comment:
Can you add some comments with examples to describe the purpose of this method?

Reply:
Nice advice